### PR TITLE
rust/rfb: use consistent key name for security_result (7.0.x backport) - v2

### PR DESCRIPTION
**Pull request by satta**
Previous PR: #11606

Link to ticket:  https://redmine.openinfosecfoundation.org/issues/7198

Changes to previous PR:
- Move documentation change to separate commit (cf #11628 for master)

### CODE REVIEW

File: rust/src/rfb/logger.rs [hunk 1]
```diff
@@ -67,12 +67,13 @@ fn log_rfb(tx: &RFBTransaction, js: &mut JsonBuilder) -> Result<(), JsonError> {
...
     }
     if let Some(security_result) = &tx.tc_security_result {
+        let result_key = "security_result";
         let _ = match security_result.status {
-            0 => js.set_string("security_result", "OK")?,
-            1 => js.set_string("security-result", "FAIL")?,
-            2 => js.set_string("security_result", "TOOMANY")?,
+            0 => js.set_string(result_key, "OK")?,
+            1 => js.set_string(result_key, "FAIL")?,
+            2 => js.set_string(result_key, "TOOMANY")?,
```
Comment (jasonish) [inline rust/src/rfb/logger.rs:74]: This is a breaking change in a patch release, I've been away, but has there been some prior discussion on this?

@catenacyber I don't see any notes in the upgrade guide for 7 to 8 as well, was this change made there as well?

Comment (catenacyber) [reply rust/src/rfb/logger.rs:74]: For me this is a fix to always use the same key

Comment (satta) [reply rust/src/rfb/logger.rs:74]: Confirmed.. While technically it is changing behaviour, it is correcting a bug.

Comment (jasonish) [reply rust/src/rfb/logger.rs:74]: Correcting a bug is not the question. It's a change of output therefore, it needs documentation on the change.

Comment (jasonish) [reply rust/src/rfb/logger.rs:74]: Actually for 7.0.x, there is a clear changelog entry and ticket describing this. I'm more concerned about 8 where there isn't as it was part of another change, so remains undocumented from what I can see. @catenacyber Can you add an upgrade note for 8?

Comment (catenacyber) [reply rust/src/rfb/logger.rs:74]: Indeed, this was an accidental fix, I propose :
https://github.com/OISF/suricata/pull/11636

**Review by jasonish — APPROVED** [review]

**Comment by victorjulien** [comment]
As #11628 is merged, can you resubmit with the proper cherry-pick line @satta ? Thanks!

**Review by catenacyber — CHANGES_REQUESTED** [review]
As Victor requested : just the right cherry-pick with commit hash from master

**Comment by satta** [comment]
Next PR: #11703

**Pull request closed without merging**